images/linux/scripts/installers/codeql-bundle.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+################################################################################
+##  File:  codeql-bundle.sh
+##  Desc:  Install the CodeQL CLI Bundle to the toolcache.
+################################################################################
+
+# Retrieve the name of the CodeQL bundle preferred by the Action (in the format codeql-bundle-YYYYMMDD).
+codeql_bundle_name="$(curl -sSL https://raw.githubusercontent.com/github/codeql-action/main/src/defaults.json | jq -r .bundleVersion)"
+# Convert the bundle name to a version number (0.0.0-YYYYMMDD).
+codeql_bundle_version="0.0.0-${codeql_bundle_name##*-}"
+
+extraction_directory="$AGENT_TOOLSDIRECTORY/CodeQL/$codeql_bundle_version/x64"
+mkdir -p "$extraction_directory"
+
+>&2 echo "Downloading CodeQL bundle $codeql_bundle_version..."
+curl -sSL "https://github.com/github/codeql-action/releases/download/$codeql_bundle_name/codeql-bundle.tar.gz" | tar -xzC "$extraction_directory"
```

Comment:
Since you download pretty huge archive, I suggest using helper function [download_with_retries](https://github.com/actions/virtual-environments/blob/1dae8d00dbbb18e26930fae977a08c90385240ef/images/linux/scripts/helpers/install.sh#L7)